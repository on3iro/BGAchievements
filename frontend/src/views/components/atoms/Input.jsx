// @flow

import styled from 'styled-components'

const Input = styled.input`
  display: block;
  border: 1px solid ${props => props.theme.inputBorderColor};
  padding: 0 ${props => props.theme.buildingUnit}px;
  width: 100%;
  height: ${props => (props.theme.buildingUnit * 5)}px;
  border-radius: ${props => (props.theme.buildingUnit * 0.5)}px;
  transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out;

  &:hover {
    background-color: ${props => props.theme.inputBackgroundColorHover};
    transition: background-color 0.2s ease-in-out;
  }

  &:focus {
    background-color: ${props => props.theme.inputBackgroundColor};
    border-color: ${props => props.theme.inputBorderColorFocus};
    outline: none;
    transition: border-color 0.2s ease-in-out;
  }

  &:disabled {
    background-color: ${props => props.theme.inputBackgroundColorDisabled};
    cursor: not-allowed;
  }
`

export {
  Input as default
}
