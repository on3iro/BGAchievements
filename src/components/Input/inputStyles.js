import { css } from 'styled-components'

const inputStyles = css`
  display: block;
  margin: 0 0 15px 0;
  padding: 5px 10px;
  width: 100%;
  height: 35px;

  border: 1px solid ${props => props.error ? props.theme.warning : props.theme.outlines};
  border-radius: 1px;
  background: ${props => props.theme.noneWhite};
  color: ${props => props.theme.text};

  outline: 0;

  &:placeholder-shown,
  &:-ms-input-placeholder {
    color: ${props => props.theme.outlines};
  }
`

export default inputStyles
