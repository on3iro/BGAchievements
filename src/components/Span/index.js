import styled from 'styled-components'

const Span = styled.span`
  display: block;
  margin: ${props => props.error ? '-10px 0 10px 5px' : '0'};
  padding-top: ${props => props.error ? '0' : '5px'}

  color: ${props => props.error ? props.theme.warning : props.theme.outlines};
  font-size: ${props => props.error ? '0.8rem' : 'inherit'};
`

export default Span
