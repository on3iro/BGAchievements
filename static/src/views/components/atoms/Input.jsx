// @flow

import styled from 'styled-components'

const Input = styled.input`
  display: block;
  border: 1px solid ${props => props.theme.borderColor};
  padding: 0 ${props => props.theme.buildingUnit}px;
  width: 100%;
  height: ${props => (props.theme.buildingUnit * 5)}px;
  border-radius: ${props => (props.theme.buildingUnit * 0.5)}px;

  &:hover {
    background-color: #efefef;
  }

  &:focus {
    background-color: ${props => props.theme.white};
    border-color: ${props => props.theme.primary};
    outline: none;
  }
`

export {
  Input as default
}
