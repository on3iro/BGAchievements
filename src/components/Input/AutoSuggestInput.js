import React from 'react'
import Autosuggest from 'react-autosuggest'
import styled from 'styled-components'

import inputStyles from './inputStyles'

export const Wrapper = styled.div`
  .react-autosuggest__container {
    position: relative;

    ul {
      margin: 20px 0 0 0 ;
      padding: 0;
    }
  }

  .react-autosuggest__input {
    ${inputStyles}
  }

  .react-autosuggest__suggestions-container {
    position: absolute;
    top: 34px;
    z-index: 10;
    overflow: auto;

    width: 100%;

    background: ${props => props.theme.noneWhite};

    border: 1px solid ${props => props.theme.outlines};
    border-top: none;
  }

  .react-autosuggest__suggestion {
    margin: 0;
    padding: 10px;
  }

  .react-autosuggest__suggestion--highlighted {
    background: ${props => props.theme.outlines};
  }
`

const AutosuggestInput = props => {
  return (
    <Wrapper>
      <Autosuggest {...props} />
    </Wrapper>
  )
}

export default AutosuggestInput
