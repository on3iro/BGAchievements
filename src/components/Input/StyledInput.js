/**
  * A simple styled controlled input element
  *
  * @namespace Input
  */

import styled from 'styled-components'

import inputStyles from './inputStyles'

const StyledInput = styled.input`${inputStyles}`

export default StyledInput
