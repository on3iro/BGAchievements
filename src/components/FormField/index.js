import React from 'react'
import PropTypes from 'prop-types'

import Input from 'components/Input'
import Span from 'components/Span'

const RenderField = ({ input, label, type, meta: { touched, error } }) => {
  return (
    <div>
      <Input
        {...input}
        placeholder={label}
        type={type}
        error={touched && error}
      />
      {touched && error && <Span error={touched && error}>{error}</Span>}
    </div>
  )
}

RenderField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  meta: PropTypes.object
}

export default RenderField
