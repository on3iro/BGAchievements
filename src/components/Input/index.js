import React from 'react'

import AutoSuggestInput from './AutoSuggestInput'
import StyledInput from './StyledInput'

const Input = props => {
  let input = <StyledInput {...props} />

  if (props.autoSuggest) {
    input = <AutoSuggestInput {...props} />
  }

  return input
}

export default Input
