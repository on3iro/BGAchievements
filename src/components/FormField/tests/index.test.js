import React from 'react'
import { shallow } from 'enzyme'

import Input from 'components/Input'
import Span from 'components/Span'

import FormField from '../index'

const renderComponent = newProps => {
  const props = {
    input: {},
    label: 'email',
    type: 'text',
    meta: {
      touched: false,
      error: false
    },
    ...newProps
  }

  const renderedWrapper = shallow(<FormField {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<FormField />', () => {
  it('should render an <Input />', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Input).length).toMatchSnapshot()
  })

  it('should render no <Span /> if it has not been touched or errored', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Span).length).toMatchSnapshot()
  })

  it('should render <Span /> when has been touched and errored', () => {
    const { renderedWrapper, props } = renderComponent({
      meta: {
        touched: true,
        error: true
      }
    })
    expect(renderedWrapper.find(Span).length).toMatchSnapshot()
  })
})
