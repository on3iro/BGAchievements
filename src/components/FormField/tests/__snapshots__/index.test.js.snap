exports[`<FormField /> should render <Span /> when has been touched and errored 1`] = `1`

exports[`<FormField /> should render an <Input /> 1`] = `1`

exports[`<FormField /> should render no <Span /> if it has not been touched or errored 1`] = `0`
